### PR TITLE
dev-lang/erlang: add command_args

### PR DESCRIPTION
**Pull request by iamdenchik**
to run epmd on different interfaces
required for rabbit cluster

Signed-off-by: Denis Pershin bazukajo@gmail.com

### CODE REVIEW

**Comment by iamdenchik** [comment]
@erikmack @mpagano pls review

File: dev-lang/erlang/files/epmd.init-r1 [hunk 1]
```diff
@@ -0,0 +1,26 @@
+#!/sbin/openrc-run
+# Copyright 1999-2018 Gentoo Foundation
+# Distributed under the terms of the Erlang Public License 1.1
+
+ZT_ADDRESS=""
+pidfile="/var/run/epmd.pid"
+command_args="${command_args:--daemon -relaxed_command_check -address 127.0.0.1}"
```
Comment (mgorny) [inline dev-lang/erlang/files/epmd.init-r1]: Why do you both install conf.d and supply a fallback? That doesn't seem to make much sense.

**Comment by iamdenchik** [comment]
@mgorny my bad. Fix https://github.com/gentoo/gentoo/pull/11830/commits/2f9b25001ecf8b30edea25d2722d5740b4296eee

**Comment by mgorny** [comment]
I'm sorry but this is currently blocked by https://bugs.gentoo.org/425928. Also, I really meant you should just set defaults in conf.d, so people can update them as they see fit, and removed it from init.d.

**Comment by iamdenchik** [comment]
> I really meant you should just set defaults in conf.d, so people can update them as they see fit, and removed it from init.d.

now I understand, sorry for my bad english

**Comment by iamdenchik** [comment]
@mgorny fix!

File: dev-lang/erlang/files/epmd.confd [hunk 2]
```diff
@@ -0,0 +1 @@
+command_args="--daemon -relaxed_command_check -address 127.0.0.1"
```
Comment (mgorny) [inline dev-lang/erlang/files/epmd.confd]: Would you mind adding a comment explaining what those default options do above, for the user's convenience?

File: dev-lang/erlang/files/epmd.init-r1 [hunk 3]
```diff
@@ -0,0 +1,26 @@
+#!/sbin/openrc-run
+# Copyright 1999-2018 Gentoo Foundation
```
Comment (mgorny) [inline dev-lang/erlang/files/epmd.init-r1]: 2019

File: dev-lang/erlang/files/epmd.init-r1 [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+# Distributed under the terms of the Erlang Public License 1.1
+
+ZT_ADDRESS=""
+pidfile="/var/run/epmd.pid"
+command_args="${command_args}"
```
Comment (mgorny) [inline dev-lang/erlang/files/epmd.init-r1]: This is redundant.

File: dev-lang/erlang/files/epmd.init-r1 [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+
+depend() {
+	need loopback
+	before sshd
+}
+
+start() {
+	ebegin "Starting Erlang Port Mapper Daemon"
+	start-stop-daemon --start --quiet \
+		--pidfile $pidfile \
```
Comment (mgorny) [inline dev-lang/erlang/files/epmd.init-r1]: While at it, please quote `${pidfile}`, and always use `${foo}` braced form for variables.

File: dev-lang/erlang/files/epmd.init-r1 (hunk 4 above)
Comment (sarnold) [reply dev-lang/erlang/files/epmd.init-r1]: Also, you can remove ZT_ADDRESS=""

Other than that it looks fine (also what mgorny said)

File: dev-lang/erlang/files/epmd.init-r1 (hunk 3 above)
Comment (iamdenchik) [reply dev-lang/erlang/files/epmd.init-r1]: fix

File: dev-lang/erlang/files/epmd.init-r1 (hunk 4 above)
Comment (iamdenchik) [reply dev-lang/erlang/files/epmd.init-r1]: fix

File: dev-lang/erlang/files/epmd.init-r1 (hunk 5 above)
Comment (iamdenchik) [reply dev-lang/erlang/files/epmd.init-r1]: fix

Comment (mgorny) [reply dev-lang/erlang/files/epmd.init-r1]: It's still not quoted.

Comment (iamdenchik) [reply dev-lang/erlang/files/epmd.init-r1]: https://github.com/gentoo/gentoo/blob/8aa2bb6a520ee169c17be513254f5099e7644a98/dev-lang/erlang/files/epmd.init-r1#L15

Comment (mgorny) [reply dev-lang/erlang/files/epmd.init-r1]: There's no quoting (`""`) there.

Comment (iamdenchik) [reply dev-lang/erlang/files/epmd.init-r1]: > quoting

sorry. fixed

**Comment by mgorny** [comment]
Thanks. I'm going to squash the commits, fix commit message, and update this for newest version in tree. Expect it merged when I test that it builds.

**Comment by graaff** [comment]
Was this tested in any way? --daemon is not an option that epmd understands (this should be -daemon) and the -- added by start-stop-daemon also breaks the epmd invocation.

**Comment by graaff** [comment]
Ah, never mind about the --, that was a misunderstanding on my part.

I'll fix the -daemon argument in dev-lang/erlang.

**Pull request closed without merging**